routers/org/home.go

```diff
@@ -119,6 +119,27 @@ func Home(ctx *context.Context) {
 		return
 	}
 
+	pinnedRepos := make([]*models.Repository, 0, 10)
+	has := false
+	for _, repo := range repos {
+		if has, err = org.IsPinnedRepoExist(repo.ID); err != nil {
+			ctx.ServerError("IsPinnedRepoExist", err)
+			return
+		}
+
+		if has {
+			pinnedRepos = append(pinnedRepos, repo)
+			repo.IsPinned = true
+		}
+	}
```

Comment:
This is highly inefficient.

This code involves doing `len(repos)` individual queries against the db. We would be better off with a single query that returns all the pinned repos for an organization, putting them in a map and then checking each repo id in that set.

We know that the number of pinned repos is in general going to be smaller than the total number of repos in an organization. (certainly it's possible that a perverse organization might decide to pin all of their private repos and none of the public ones but that would be rare.)

Reply:
Hmm, It's really a hard thing to balance memory usage and run-efficiency here. How about this? Thanks.  73ab59623fb88edf913c6793c4be370bd4c160bc